src/test/java/org/opensearch/searchrelevance/experiment/ExperimentOptionsForHybridSearchTests.java

```diff
@@ -241,4 +241,42 @@ public void testGetParameterCombinations_withPreciseIncrements_shouldIncludeAllW
             assertTrue("Missing expected weight: " + expected, foundWeight);
         }
     }
+
+    public void testGetParameterCombinations_weightsAreRoundedToOneDecimal() {
```

Comment:
Can we add one more test case (or change this one) where we compare generated weights with constant list of correct values (11 value pairs in a range from 0.0/1.0 to 1.0/0.0). This will make sure we not only have correct values in each pair, but also have only valid pairs and don't miss any valid pair